Kic plugin howto (#1200)
Kong/developer.konghq.com: app/_how-tos/kic-plugin-authentication.md

    ```bash
    echo '
    apiVersion: v1
    kind: Secret
    metadata:
      name: consumer-1-basic-auth
      namespace: kong
      labels:
        konghq.com/credential: basic-auth
    stringData:
        username: consumer-1
        password: consumer-1-password
    ' | kubectl apply -f -
    ```

1. Create a secret to add a `key-auth` credential for `consumer-2`:

    ```bash
    echo '
    apiVersion: v1
    kind: Secret
    metadata:
      name: consumer-2-key-auth
      namespace: kong
      labels:
        konghq.com/credential: key-auth
    stringData:
      key: consumer-2-password
    ' | kubectl apply -f -
    ```

1.  Create a consumer named `consumer-1`:

{% entity_example %}
type: consumer
data:
  username: consumer-1
  credentials:
    - consumer-1-basic-auth
  
indent: 4
{% endentity_example %}

1.  Create a consumer named `consumer-2`:

{% entity_example %}
type: consumer
data:
  username: consumer-2
  credentials:
    - consumer-2-key-auth
  
indent: 4
{% endentity_example %}

## Secure the service

Once the Consumers and credentials are created, you can add authentication plugins to your Service.

> First, create a `key-auth` plugin. Notice the `anonymous` configuration option, which means "if no credentials match, assign the consumer named `anonymous` to the request":

Reviewer: ```suggestion
First, create a [`key-auth`](/plugins/key-auth/) plugin. Notice the `anonymous` configuration option, which means that if no credentials match, the consumer named `anonymous` is assigned to the request:
```

{% entity_example %}
type: plugin
data:
  name: key-auth
  config:
    anonymous: anonymous
  
  service: echo
  skip_annotate: true
{% endentity_example %}

> As Consumer 2 is using `basic-auth`, we also need to create a `basic-auth` plugin:

Reviewer: ```suggestion
As Consumer 2 is using `basic-auth`, we also need to create a [`basic-auth`](/plugins/basic-auth/) plugin:
```

{% entity_example %}
type: plugin
data:
  name: basic-auth
  config:
    anonymous: anonymous
  
  service: echo
  other_plugins: key-auth
{% endentity_example %}

> ## Create an anonymous consumer

Reviewer: ```suggestion
## Create an anonymous Consumer
```

> Your endpoints are now secure, but neither consumer can access the endpoint when providing valid credentials. This is because each plugin will verify the consumer using it’s own authentication method.

Reviewer: ```suggestion
Your endpoints are now secure, but neither Consumer can access the endpoint when providing valid credentials. This is because each plugin will verify the Consumer using it’s own authentication method.
```

> To allow multiple authentication methods, create an anonymous consumer which is the default user if no valid credentials are provided:

Reviewer: ```suggestion
To allow multiple authentication methods, create an anonymous Consumer which is the default user if no valid credentials are provided:
```

{% entity_example %}
type: consumer
data:
  username: anonymous
{% endentity_example %}

> All requests to the API will now succeed as the anonymous consumer is being used as a default.

Reviewer: ```suggestion
All requests to the API will now succeed as the anonymous Consumer is being used as a default.
```
